src/v/net/probes.cc

```diff
@@ -20,13 +21,13 @@
 
 namespace net {
 void server_probe::setup_metrics(
-  ss::metrics::metric_groups& mgs, const char* proto) {
+  ss::metrics::metric_groups& mgs, std::string_view proto) {
     namespace sm = ss::metrics;
     auto aggregate_labels = config::shard_local_cfg().aggregate_metrics()
                               ? std::vector<sm::label>{sm::shard_label}
                               : std::vector<sm::label>{};
     mgs.add_group(
-      prometheus_sanitize::metrics_name(proto),
+      prometheus_sanitize::metrics_name(proto.data()),
```

Comment:
This is safe to do because `server_configuration::name` is a `sstring, but in isolation this is not safe. The `sstring` constructor for this overload calls `strlen` and `string_view` pointers aren't null terminated:
```
basic_sstring(const char* x) : basic_sstring(reinterpret_cast<const char_type*>(x), std::strlen(x)) {}
```

I'd explicitly call the `string_view` constructor: `ss::sstring{proto}` as there's probably no implicit conversion.

Reply:
Good tip, I'll make the change in the next push after I figure out why raft unit tests are failing with my changes.

Reply:
Done.